plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftCastPushdown.java

```diff
@@ -0,0 +1,543 @@
+                    .failure()
+                    .hasMessageContaining("Value out of range for 4 bytes");
+
+            assertThat(query(sessionWithoutPushdown, "SELECT id, CAST(c_decimal_25_2 AS INTEGER) from %s WHERE id = 12".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Cannot cast '2147483648.49' to INTEGER");
+            assertThat(query("SELECT id, CAST(c_decimal_25_2 AS INTEGER) from %s WHERE id = 12".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Value out of range for 4 bytes");
+
+            assertThat(query(sessionWithoutPushdown, "SELECT id, CAST(c_decimal_25_2 AS BIGINT) from %s WHERE id = 13".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Cannot cast '-9223372036854775809.49' to BIGINT");
+            assertThat(query("SELECT id, CAST(c_decimal_25_2 AS BIGINT) from %s WHERE id = 13".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Value out of range for 8 bytes");
+
+            assertThat(query(sessionWithoutPushdown, "SELECT id, CAST(c_decimal_25_2 AS BIGINT) from %s WHERE id = 14".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Cannot cast '-9223372036854775808.94' to BIGINT");
+            assertThat(query("SELECT id, CAST(c_decimal_25_2 AS BIGINT) from %s WHERE id = 14".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Value out of range for 8 bytes");
+
+            assertThat(query(sessionWithoutPushdown, "SELECT id, CAST(c_decimal_25_2 AS BIGINT) from %s WHERE id = 15".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Cannot cast '9223372036854775807.94' to BIGINT");
+            assertThat(query("SELECT id, CAST(c_decimal_25_2 AS BIGINT) from %s WHERE id = 15".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Value out of range for 8 bytes");
+
+            assertThat(query(sessionWithoutPushdown, "SELECT id, CAST(c_decimal_25_2 AS BIGINT) from %s WHERE id = 16".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Cannot cast '9223372036854775808.49' to BIGINT");
+            assertThat(query("SELECT id, CAST(c_decimal_25_2 AS BIGINT) from %s WHERE id = 16".formatted(table.getName())))
+                    .failure()
+                    .hasMessageContaining("Value out of range for 8 bytes");
+        }
+    }
+
+    @Test
+    public void testCastWithForcedMappedUnsupportedType()
+    {
+        // c_timetz is not supported by default by trino. This is forced mapped to varchar.
+        assertThat(query("SELECT CAST(c_timetz AS smallint) FROM %s".formatted(leftTable())))
```

Comment:
can this move under `failCast()` ?

Reply:
This is a special case for `.put("unsupported-type-handling", "CONVERT_TO_VARCHAR")`

Reply:
Moved to `invalidCast`